File: app/src/main/java/de/markusfisch/android/shadereditor/activity/LoadSampleActivity.java
package de.markusfisch.android.shadereditor.activity;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.support.v4.app.Fragment;

import de.markusfisch.android.shadereditor.fragment.LoadSampleFragment;

public class LoadSampleActivity extends AbstractContentActivity {
	public static final String NAME = "name";
	public static final String RESOURCE_ID = "resource_id";
	public static final String THUMBNAIL_ID = "thumbnail_id";
	public static final String QUALITY = "quality";

	public static void setSampleResult(
			Activity activity,
			String name,
			int resId,
			int thumbId,
			float quality) {
		Bundle bundle = new Bundle();
		bundle.putString(NAME, name);
		bundle.putInt(RESOURCE_ID, resId);
		bundle.putInt(THUMBNAIL_ID, thumbId);
		bundle.putFloat(QUALITY, quality);

		Intent data = new Intent();
		data.putExtras(bundle);

		activity.setResult(RESULT_OK, data);
	}

	@Override
	protected Fragment defaultFragment() {
		return new LoadSampleFragment();
	}
}

File: app/src/main/java/de/markusfisch/android/shadereditor/activity/LoadSampleActivity.kt
package de.markusfisch.android.shadereditor.activity

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import androidx.fragment.app.Fragment
import de.markusfisch.android.shadereditor.fragment.LoadSampleFragment

class LoadSampleActivity : AbstractContentActivity() {

    companion object {
        const val NAME = "name"
        const val RESOURCE_ID = "resource_id"
        const val THUMBNAIL_ID = "thumbnail_id"
        const val QUALITY = "quality"

        @JvmStatic
        fun setSampleResult(
            activity: Activity, name: String, resId: Int, thumbId: Int, quality: Float
        ) {
            val bundle = Bundle().apply {
                putString(NAME, name)
                putInt(RESOURCE_ID, resId)
                putInt(THUMBNAIL_ID, thumbId)
                putFloat(QUALITY, quality)
            }

            val data = Intent().apply {
                putExtras(bundle)
            }

            activity.setResult(Activity.RESULT_OK, data)
        }
    }

    override fun defaultFragment(): Fragment {
        return LoadSampleFragment()
    }
}